src/coreclr/vm/jitinterface.cpp

```diff
@@ -6504,6 +6504,33 @@ bool CEEInfo::isIntrinsic(CORINFO_METHOD_HANDLE ftn)
     return ret;
 }
 
+bool CEEInfo::notifyMethodInfoUsage(CORINFO_METHOD_HANDLE ftn)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    bool ret = true;
+
+    JIT_TO_EE_TRANSITION_LEAF();
+
+    _ASSERTE(ftn);
+
+#ifdef PROFILING_SUPPORTED
+    if (CORProfilerEnableRejit())
+    {
+        // If ReJIT is around we'd better be careful with observations during inlining.
+        ret = false;
+    }
+#endif // PROFILING_SUPPORTED
+
+    EE_TO_JIT_TRANSITION_LEAF();
```

Comment:
```suggestion
    JIT_TO_EE_TRANSITION();

    _ASSERTE(ftn);

#ifdef FEATURE_REJIT
    MethodDesc *pCallee = GetMethod(ftn);
    MethodDesc *pCaller = m_pMethodBeingCompiled;
    pCallee->GetModule()->AddInlining(pCaller, pCallee);
#endif // PROFILING_SUPPORTED

    EE_TO_JIT_TRANSITION();
    
    return true;
```
Re-JIT can be enabled mid-flight. All code needs to be Re-JIT aware by default.

Reply:
Ah, yeah I had that in mind, but I thought that for the mid-flight case the only solution is to never do "no-return" logic in JIT at all.